test/testutil/util.go

```
@@ -32,6 +32,14 @@ func MustGetTestData[T any](file string) *T {
 	return &data
 }
 
+// MustUnmarshalFromFile reads testdata and unmarshals it to the given type, panicking if an error occurs.
+func MustUnmarshalFromFile(file string, out any) {
+	err := json.Unmarshal(ReadFixture(file), out)
```

I believe that any can also get a pointer.

---

any is interface{}, so it can hold any data type.

---

We should use decoder and reject unknown fields.  It's very easy for test data to cover the wrong case if we silently ignore extra fields.